build.sbt

```diff
@@ -93,9 +93,11 @@ lazy val tyrian =
     .jsSettings(
       commonJsSettings,
       libraryDependencies ++= Seq(
-        "org.scala-js"  %%% "scalajs-dom"        % Dependancies.scalajsDomVersion,
-        "org.typelevel" %%% "cats-effect-kernel" % Dependancies.catsEffect,
-        "co.fs2"        %%% "fs2-core"           % Dependancies.fs2
+        "org.scala-js"  %%% "scalajs-dom"               % Dependancies.scalajsDomVersion,
+        "org.typelevel" %%% "cats-effect-kernel"        % Dependancies.catsEffect,
+        "co.fs2"        %%% "fs2-core"                  % Dependancies.fs2,
+        ("org.scala-js" %%% "scalajs-java-securerandom" % Dependancies.scalajsJavaSecureRandom)
+          .cross(CrossVersion.for3Use2_13)
```

Comment:
AFAICT this dependency is not needed by tyrian actually. It is needed by indigo perhaps, but not tyrian core.

For the reasons described in this PR I recommend against adding this dependency.
- https://github.com/typelevel/cats-effect/pull/2945

Reply:
In other words, you probably want to add it here :)
https://github.com/PurpleKingdomGames/tyrian/blob/783d6b0afef27bfe9955aa9ac851c3bbbcef9b35/build.sbt#L120-L130

Reply:
You are not wrong, but you're not quite right either. I'm investigating. It's not clear to me why Indigo needs it either - other than compilation fails, unless it's just because of the Scala.js versions being different as I originally suspected.

Regarding your recommendation to use the Cats one, yes I probably could do that for Tyrian, if Tyrian needs it at all.

Reply:
Well, I'm going by the CI logs, and they all seem to point here:
https://github.com/PurpleKingdomGames/indigo/blob/e4a55b6b4cb93d6dd2b162a0ab2773d097d79b8c/indigo/indigo/src/main/scala/indigo/shared/subsystems/SubSystemsRegister.scala#L34

```
[error] Referring to non-existent class java.security.SecureRandom
[error]   called from private java.util.UUID$.csprng$lzycompute()java.util.Random
[error]   called from private java.util.UUID$.csprng()java.util.Random
[error]   called from java.util.UUID$.randomUUID()java.util.UUID
[error]   called from static java.util.UUID.randomUUID()java.util.UUID
[error]   called from private indigo.shared.subsystems.SubSystemsRegister.initialiseSubSystem(indigo.shared.subsystems.SubSystem)indigo.shared.Outcome
[error]   called from private indigo.shared.subsystems.SubSystemsRegister.register$$anonfun$1(indigo.shared.subsystems.SubSystem)indigo.shared.Outcome
[error]   called from indigo.shared.subsystems.SubSystemsRegister.register(scala.collection.immutable.List)scala.collection.immutable.List
[error]   called from private indigo.IndigoGame.indigoGame(indigo.BootResult)indigo.gameengine.GameEngine
[error]   called from indigo.IndigoGame.ready(java.lang.String,scala.collection.immutable.Map)indigo.gameengine.GameEngine
[error]   called from example.game.MyAwesomeGame.ready(java.lang.String,scala.collection.immutable.Map)indigo.gameengine.GameEngine
[error]   called from indigo.GameLauncher.launch()void
[error]   called from example.game.MyAwesomeGame.launch()void
[error]   called from indigo.GameLauncher.$js$exported$meth$launch()java.lang.Object
[error]   called from example.game.MyAwesomeGame.$js$exported$meth$launch()java.lang.Object
[error]   exported to JavaScript with @JSExport
[error] involving instantiated classes:
[error]   java.util.UUID$
[error]   indigo.shared.subsystems.SubSystemsRegister
[error]   example.game.MyAwesomeGame
[error] Referring to non-existent class java.security.SecureRandom
[error]   called from private java.util.UUID$.csprng$lzycompute()java.util.Random
[error]   called from private java.util.UUID$.csprng()java.util.Random
[error]   called from java.util.UUID$.randomUUID()java.util.UUID
[error]   called from static java.util.UUID.randomUUID()java.util.UUID
[error]   called from private indigo.shared.subsystems.SubSystemsRegister.initialiseSubSystem(indigo.shared.subsystems.SubSystem)indigo.shared.Outcome
[error]   called from private indigo.shared.subsystems.SubSystemsRegister.register$$anonfun$1(indigo.shared.subsystems.SubSystem)indigo.shared.Outcome
[error]   called from indigo.shared.subsystems.SubSystemsRegister.register(scala.collection.immutable.List)scala.collection.immutable.List
[error]   called from private indigo.IndigoGame.indigoGame(indigo.BootResult)indigo.gameengine.GameEngine
[error]   called from indigo.IndigoGame.ready(java.lang.String,scala.collection.immutable.Map)indigo.gameengine.GameEngine
[error]   called from example.game.MyAwesomeGame.ready(java.lang.String,scala.collection.immutable.Map)indigo.gameengine.GameEngine
[error]   called from indigo.GameLauncher.launch()void
[error]   called from example.game.MyAwesomeGame.launch()void
[error]   called from indigo.GameLauncher.$js$exported$meth$launch()java.lang.Object
[error]   called from example.game.MyAwesomeGame.$js$exported$meth$launch()java.lang.Object
[error]   exported to JavaScript with @JSExport
[error] involving instantiated classes:
[error]   java.util.UUID$
[error]   indigo.shared.subsystems.SubSystemsRegister
[error]   example.game.MyAwesomeGame
[error] Referring to non-existent method constructor java.security.SecureRandom.<init>()void
[error]   called from private java.util.UUID$.csprng$lzycompute()java.util.Random
[error]   called from private java.util.UUID$.csprng()java.util.Random
[error]   called from java.util.UUID$.randomUUID()java.util.UUID
[error]   called from static java.util.UUID.randomUUID()java.util.UUID
[error]   called from private indigo.shared.subsystems.SubSystemsRegister.initialiseSubSystem(indigo.shared.subsystems.SubSystem)indigo.shared.Outcome
[error]   called from private indigo.shared.subsystems.SubSystemsRegister.register$$anonfun$1(indigo.shared.subsystems.SubSystem)indigo.shared.Outcome
[error]   called from indigo.shared.subsystems.SubSystemsRegister.register(scala.collection.immutable.List)scala.collection.immutable.List
[error]   called from private indigo.IndigoGame.indigoGame(indigo.BootResult)indigo.gameengine.GameEngine
[error]   called from indigo.IndigoGame.ready(java.lang.String,scala.collection.immutable.Map)indigo.gameengine.GameEngine
[error]   called from example.game.MyAwesomeGame.ready(java.lang.String,scala.collection.immutable.Map)indigo.gameengine.GameEngine
[error]   called from indigo.GameLauncher.launch()void
[error]   called from example.game.MyAwesomeGame.launch()void
[error]   called from indigo.GameLauncher.$js$exported$meth$launch()java.lang.Object
[error]   called from example.game.MyAwesomeGame.$js$exported$meth$launch()java.lang.Object
[error]   exported to JavaScript with @JSExport
[error] involving instantiated classes:
[error]   java.util.UUID$
[error]   indigo.shared.subsystems.SubSystemsRegister
[error]   example.game.MyAwesomeGame
[error] Referring to non-existent class java.security.SecureRandom
[error]   called from private java.util.UUID$.csprng$lzycompute()java.util.Random
[error]   called from private java.util.UUID$.csprng()java.util.Random
[error]   called from java.util.UUID$.randomUUID()java.util.UUID
[error]   called from static java.util.UUID.randomUUID()java.util.UUID
[error]   called from private indigo.shared.subsystems.SubSystemsRegister.initialiseSubSystem(indigo.shared.subsystems.SubSystem)indigo.shared.Outcome
[error]   called from private indigo.shared.subsystems.SubSystemsRegister.register$$anonfun$1(indigo.shared.subsystems.SubSystem)indigo.shared.Outcome
[error]   called from indigo.shared.subsystems.SubSystemsRegister.register(scala.collection.immutable.List)scala.collection.immutable.List
[error]   called from private indigo.IndigoGame.indigoGame(indigo.BootResult)indigo.gameengine.GameEngine
[error]   called from indigo.IndigoGame.ready(java.lang.String,scala.collection.immutable.Map)indigo.gameengine.GameEngine
[error]   called from example.game.MyAwesomeGame.ready(java.lang.String,scala.collection.immutable.Map)indigo.gameengine.GameEngine
[error]   called from indigo.GameLauncher.launch()void
[error]   called from example.game.MyAwesomeGame.launch()void
[error]   called from indigo.GameLauncher.$js$exported$meth$launch()java.lang.Object
[error]   called from example.game.MyAwesomeGame.$js$exported$meth$launch()java.lang.Object
[error]   exported to JavaScript with @JSExport
[error] involving instantiated classes:
[error]   java.util.UUID$
[error]   indigo.shared.subsystems.SubSystemsRegister
[error]   example.game.MyAwesomeGame
```

Reply:
My apologies, you are indeed correct. I had completely forgotten about that one use of UUID in the SubsystemsRegister.

Reply:
I may yet just remove the UUID code entirely from Indigo and release them both together...

Reply:
scalajs-java-securerandom has indeed been removed.